src/test/worker/worker_server_proxy.py

```diff
@@ -0,0 +1,118 @@
+from ja.worker.proxy.proxy import IWorkerServerProxy
+from ja.server.database.types.work_machine import WorkMachine, WorkMachineState, WorkMachineResources
+from ja.common.job import Job, JobStatus, JobPriority, JobSchedulingConstraints
+from ja.common.docker_context import DockerContext, DockerConstraints
+from ja.common.message.server import ServerResponse
+from ja.common.work_machine import ResourceAllocation
+from typing import Dict
+import copy
+from unittest import TestCase
+from test.abstract import skipIfAbstract
+
+
+class WorkerServerProxyDummy(IWorkerServerProxy):
+    def __init__(self, wmcs: Dict[str, WorkMachine], jobs: Dict[str, Job]):
+        self._wmcs: Dict[str, WorkMachine] = copy.deepcopy(wmcs)
+        self._jobs: Dict[str, Job] = copy.deepcopy(jobs)
+
+    def register_self(self, uid: str, work_machine_resources: WorkMachineResources) -> ServerResponse:
+        if uid in self._wmcs:
+            return ServerResponse("work machine already exist!", False)
+        self._wmcs[uid] = WorkMachine(
+            uid="workmachine-1",
```

Comment:
This should use the uid argument instead of a string literal.

Reply:
the code seem to have some assert false tests, do you have some specific params you want me to check?

Reply:
Sorry, it seems I made a mistake when reviewing your PR.
It seems your PR already has the tests that I wanted and I just didn't see them.